airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryAvroSerializedBuffer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.bigquery;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.functional.CheckedBiFunction;
+import io.airbyte.integrations.base.AirbyteStreamNameNamespacePair;
+import io.airbyte.integrations.destination.bigquery.formatter.BigQueryRecordFormatter;
+import io.airbyte.integrations.destination.record_buffer.BufferStorage;
+import io.airbyte.integrations.destination.record_buffer.SerializableBuffer;
+import io.airbyte.integrations.destination.s3.avro.AvroSerializedBuffer;
+import io.airbyte.integrations.destination.s3.avro.S3AvroFormatConfig;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import io.airbyte.protocol.models.AirbyteStream;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.io.IOException;
+import java.util.concurrent.Callable;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import org.apache.avro.Schema;
+import org.apache.avro.file.CodecFactory;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * This class differs from {@link AvroSerializedBuffer} in that 1) the Avro schema can be customized
+ * by the caller, and 2) the message is formatted by {@link BigQueryRecordFormatter}. In this way,
+ * this buffer satisfies the needs of both the standard and the denormalized BigQuery destinations.
+ */
```

Comment:
Shouldn't we introduce a common concept to handle this though?

If I understand correctly, this sounds like the [CsvSheetGenerator](https://github.com/airbytehq/airbyte/blob/b25c0de6839bac755c3cc02a6cc36e0d0b188b65/airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/csv/CsvSheetGenerator.java#L17)?

Reply:
At this point, only BigQuery needs a separate implementation, because the BigQuery schema is a bit hacky. I doubt if other destinations will ever need this. So I think something more generic is not necessary. Maybe let's wait until there is another case use?

Reply:
👌